src/frontend/src/components/NavBarV2/index.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from "react";
+import ENV from "src/common/constants/ENV";
+import Exclamation from "src/common/icons/IconExclamation.svg";
+import CloseIcon from "src/common/images/close-icon.svg";
+import HeaderLogo from "src/common/images/gen-epi-logo.svg";
+import { useUserInfo } from "src/common/queries/auth";
+import { ROUTES } from "src/common/routes";
+import UserMenu from "./components/UserMenu";
+import {
+  AnnouncementBanner,
+  AnnouncementText,
+  AnnouncementTextBold,
+  Bar,
+  ButtonLink,
+  HeaderContainer,
+  HeaderLogoContainer,
+  HeaderMaxWidthContainer,
+  HeaderTopContainer,
+  HeaderTopLinks,
+  MobileNavClose,
+  MobileNavCloseContainer,
+  MobileNavLink,
+  MobileNavLinkContainer,
+  MobileNavSeparator,
+  MobileNavToggle,
+  MobileNavTray,
+  OrgSplash,
+  TextLink,
+} from "./style";
+
+export default function NavBarV2(): JSX.Element {
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  function toggleMobileNav() {
+    setMenuOpen(!menuOpen);
+  }
+
+  const { API_URL } = ENV;
+
+  const { data } = useUserInfo();
+
+  const group = data?.group;
+
+  const orgElements = <React.Fragment>{group?.name}</React.Fragment>;
+
+  function hasOrg(): JSX.Element | null {
+    if (group === undefined) {
+      return null;
+    } else {
+      return orgElements;
+    }
+  }
+
+  const orgSplash = hasOrg();
+
+  const user = data?.user;
+
+  let MobileSignInLink;
+  let SignInLink;
+
+  if (user) {
+    MobileSignInLink = (
+      <MobileNavLink
+        href={ROUTES.UPLOAD_STEP1}
+        style={menuOpen ? { opacity: "1" } : { opacity: "0" }}
+        rel="noreferrer"
+        aria-label="Upload Data"
+      >
+        Upload
+      </MobileNavLink>
+    );
+
+    SignInLink = <ButtonLink href={ROUTES.UPLOAD_STEP1}>Upload</ButtonLink>;
+  } else {
+    MobileSignInLink = (
+      <MobileNavLink
+        href={API_URL + ROUTES.LOGIN}
+        style={menuOpen ? { opacity: "1" } : { opacity: "0" }}
+        rel="noreferrer"
+        aria-label="Log into CZI GEN EPI"
+      >
+        Sign In
+      </MobileNavLink>
+    );
+    SignInLink = <ButtonLink href={API_URL + ROUTES.LOGIN}>Sign in</ButtonLink>;
+  }
+
+  return (
+    <HeaderContainer>
+      <AnnouncementBanner>
+        <AnnouncementText>
+          <Exclamation />
+          <AnnouncementTextBold>Looking for Aspen?</AnnouncementTextBold>
+          &nbsp;You&apos;re in the right spot. As of December, our new name is
+          CZ GEN EPI.
+        </AnnouncementText>
+      </AnnouncementBanner>
+      <HeaderMaxWidthContainer>
+        <HeaderTopContainer>
+          <HeaderLogoContainer href={data ? ROUTES.DATA : ROUTES.HOMEPAGE}>
+            <HeaderLogo />
+            {orgSplash ? <OrgSplash>{orgSplash}</OrgSplash> : null}
+          </HeaderLogoContainer>
+          <HeaderTopLinks>
+            {user ? null : (
+              <TextLink href={ROUTES.RESOURCES} target="_blank">
```

Comment:
for any links that are opening in a new tab, consider using `NewTabLink`. This also prevents the new tab from grabbing old browsing context or referrer info from our site.